akd/src/append_only_zks.rs

```diff
@@ -241,9 +241,7 @@ impl Azks {
         }
 
         while !hash_q.is_empty() {
-            let (next_node_label, _) = hash_q
-                .pop()
-                .ok_or(AzksError::PopFromEmptyPriorityQueue(self.latest_epoch))?;
+            let (next_node_label, _) = hash_q.pop().unwrap();
```

Comment:
unwrap() is going to panic if it fails, rather than bubble up the error in an expected format. i.e. panic will terminate the app completely

Reply:
Also @Jasleen1 do we need the priority queue in the end here? Dropping it would let us also drop a crate dependency :)

Reply:
`PopFromEmptyPriorityQueue` implies the queue is empty but there is already an `isEmpty` check right before. 

Could `unwrap` fail here?

Reply:
So maybe we could just rewrite as a ```while let Ok((next_node_label, _)) = ...``` statement, an avoid the is empty check all together.

Reply:
But yes I believe it can't fail, there's just better syntax to be used I think

Reply:
I like the `while let` approach, updated!